fl4health/feature_alignment/tab_features_info_encoder.py

```diff
@@ -0,0 +1,130 @@
+import json
+from typing import Dict, List, Optional, Union
+
+import pandas as pd
+from cyclops.process.feature.feature import TabularFeatures
+from flwr.common.typing import Scalar
+from sklearn.feature_extraction.text import CountVectorizer
+
+from fl4health.feature_alignment.tabular_feature import MetaData, TabularFeature
+from fl4health.feature_alignment.tabular_type import TabularType
+
+
+class TabularFeaturesInfoEncoder:
+    """
+    This class encodes all the information required to perform feature
+    alignment on tabular datasets.
+
+    Parameters
+    ----------
+    tabular_features: List[TabularFeature]
+        List of all tabular features.
+    tabular_targets: List[TabularFeature]
+        List of all targets. (Note: targets are not included in tabular_features)
+    """
+
+    def __init__(self, tabular_features: List[TabularFeature], tabular_targets: List[TabularFeature]) -> None:
+        self.tabular_features = sorted(tabular_features, key=TabularFeature.get_feature_name)
+        self.tabular_targets = sorted(tabular_targets, key=TabularFeature.get_feature_name)
+
+    def get_tabular_features(self) -> List[TabularFeature]:
+        return self.tabular_features
+
+    def get_tabular_targets(self) -> List[TabularFeature]:
+        return self.tabular_targets
+
+    def get_feature_columns(self) -> List[str]:
+        return sorted([feature.get_feature_name() for feature in self.tabular_features])
```

Comment:
I'm probably missing something, but is there any reason why this list needs to be sorted?

Reply:
A similar question might apply to `get_target_columns`

Reply:
I guess it doesn't have to be sorted strictly speaking... It was a remanent from the previous iteration of the design. I kept it here because I thought it would make testing easier.